pdata/pcommon/generated_primitive_slice.go

```diff
@@ -0,0 +1,197 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by "model/internal/cmd/pdatagen/main.go". DO NOT EDIT.
+// To regenerate this file run "go run model/internal/cmd/pdatagen/main.go".
+
+package pcommon
+
+import "go.opentelemetry.io/collector/pdata/internal"
+
+// ByteSlice represents a []byte slice.
+// The instance of ByteSlice can be assigned to multiple objects since it's immutable.
+//
+// Must use NewByteSlice function to create new instances.
+// Important: zero-initialized instance is not valid for use.
+type ByteSlice internal.ByteSlice
+
+// Deprecated: [0.59.0] Use ByteSlice instead.
+type ImmutableByteSlice = ByteSlice
+
+func (ms ByteSlice) getOrig() *[]byte {
+	return internal.GetOrigByteSlice(internal.ByteSlice(ms))
+}
+
+// NewByteSlice creates a new empty ByteSlice.
+func NewByteSlice() ByteSlice {
+	orig := []byte(nil)
+	return ByteSlice(internal.NewByteSlice(&orig))
+}
+
+// NewImmutableByteSlice creates a new ByteSlice by copying the provided []byte slice.
+// Deprecated: [0.59.0] Use New{structName}() and {structName}.CopyFromRaw([]byte) instead
+func NewImmutableByteSlice(orig []byte) ByteSlice {
+	return ByteSlice(internal.NewByteSlice(&orig))
+}
+
+// AsRaw returns a copy of the []byte slice.
+func (ms ByteSlice) AsRaw() []byte {
+	orig := *ms.getOrig()
+	if len(orig) == 0 {
+		return nil
+	}
+	copyOrig := make([]byte, len(orig))
+	copy(copyOrig, orig)
+	return copyOrig
+}
+
+// CopyFromRaw copies raw []byte into the slice ByteSlice.
+func (ms ByteSlice) CopyFromRaw(val []byte) {
```

Comment:
Still not sure about the `Copy` word in this func name.

Reply:
If we change `AsRaw` to copy the data as well, we should make consistent naming. Let's go without `Copy` prefix